modules/social_features/social_activity/social_activity.module

```diff
@@ -234,3 +235,57 @@ function social_activity_views_post_build(ViewExecutable $view) {
     }
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_activity_preprocess_block(&$variables) {
```

Comment:
Just a question out of curiosity, since we have views in place. Doesn't views allow us to use the lazy building already by default? It seems we are adding a hell of a lot of alteration just for this, while i assume this core behavior to work and I don't see any explanation in code why its necessary to use this preprocess. 
Also looking at the `viewsLazyBuild` the only thing we do there are things views should take care of. 

Did you guys check that scenario if core isn't able to do this easier for us?

Reply:
Hi, @ronaldtebrake we investigated it with a team before and unfortunately Drupal doesn't have a solution for that.

Reply:
Hi @dmytriikaiun alright, thanks for the explanation. While im thinking about it, I think this is actually very valuable to add to core in this case. So I can agree that this is necessary for now and we might be able to port this to core inside the rendering of views in a follow-up. Let's make core better with your awesome work :D !
So for me it's good to merge! :)